### PR TITLE
Add RX numbers to CPN models list (2.3)

### PR DESCRIPTION
**Pull request by elecpower**

### CODE REVIEW

File: companion/src/modelslist.cpp [hunk 1]
```diff
@@ -710,10 +720,46 @@ void TreeModel::refresh()
...
 }
+
+bool TreeModel::isModelIdUnique(unsigned modelIdx)
+{
+  int cnt = 0;
+  for (unsigned i=0; i<radioData->models.size(); i++) {
+    ModelData & model = radioData->models[i];
+    for (unsigned j=0; j<CPN_MAX_MODULES; j++) {
+      if (!model.moduleData[j].protocol == PULSES_OFF && model.moduleData[j].modelId == modelIdx) {
+        cnt++;
```
Comment (bsongis) [inline companion/src/modelslist.cpp]: you may return false from here to avoid looping on all models

Comment (elecpower) [reply companion/src/modelslist.cpp]: @bsongis yes will speed up. In testing the change I also found a bug and will fix both in the next commit.

**Comment by elecpower** [comment]
@bsongis after retesting there isn't a bug, well I hope not. So just exit as soon as 1st duplicate found.

File: companion/src/mdichild.cpp [hunk 2]
```diff
@@ -543,6 +542,14 @@ void MdiChild::initModelsList()
     ui->modelsList->setIndentation(0);
   }
   refresh();
+
+  if (firmware->getCapability(Capability::HasModelCategories)) {
+    ui->modelsList->header()->resizeSection(0, ui->modelsList->header()->sectionSize(0) * 2);   // pad out categories and model names
+    }
```
Comment (bsongis) [inline companion/src/mdichild.cpp]: indentation issue there!

File: companion/src/modelslist.cpp [hunk 3]
```diff
@@ -710,10 +720,53 @@ void TreeModel::refresh()
...
 }
+
+bool TreeModel::isModelIdUnique(unsigned modelIdx)
+{
+  int cnt = 0;
+  for (unsigned i=0; i<radioData->models.size(); i++) {
+    ModelData & model = radioData->models[i];
+    if (!model.isEmpty()) {
+      for (unsigned j=0; j<CPN_MAX_MODULES; j++) {
+        if (!model.moduleData[j].protocol == PULSES_OFF && model.moduleData[j].modelId == modelIdx) {
```
Comment (bsongis) [inline companion/src/modelslist.cpp]: `model.moduleData[j].protocol != PULSES_OFF`

File: companion/src/modelslist.cpp [hunk 4]
```diff
@@ -710,10 +720,53 @@ void TreeModel::refresh()
...
+          if (cnt < 1) {
```
Comment (bsongis) [inline companion/src/modelslist.cpp]:
```
if (cnt++ > 0) {
  return false;
}
```

**Comment by bsongis** [comment]
Thanks!

**Pull request merged**